packages/compiler/src/render3/view/compiler.ts

```diff
@@ -435,13 +436,54 @@ function selectorsFromGlobalMetadata(
 }
 
 function prepareQueryParams(query: R3QueryMetadata, constantPool: ConstantPool): o.Expression[] {
-  const parameters = [getQueryPredicate(query, constantPool), o.literal(query.descendants)];
+  const parameters = [getQueryPredicate(query, constantPool), o.literal(toQueryFlags(query))];
   if (query.read) {
     parameters.push(query.read);
   }
   return parameters;
 }
 
+/**
+ * A set of flags to be used with Queries.
+ *
+ * NOTE: Ensure changes here are reflected in `packages/compiler/src/render3/view/compiler.ts`
+ */
+export const enum QueryFlags {
+  /**
+   * No flags
+   */
+
+  none = 0,
```

Comment:
Small indentation update (rm line before `none`, add one after):
```suggestion
  none = 0,

```

Reply:
Why line after? I think it is easier to read. Other places have line breaks like that as well.

Reply:
It just looks like the comment is not "attached" to the `none = 0` string (since there is a line in between a comment and an expression).

Reply:
yes, that one (break between comment and declaration) I removed. But you are suggesting to remove the empty line after. That one I think should stay.

Reply:
Yes, I agree that we should keep an empty line after 👍